pkg/sql/sem/builtins/generator_builtins.go

```diff
@@ -147,13 +147,11 @@ var generators = map[string]builtinDefinition{
 			},
 			spanKeyIteratorType,
 			func(ctx context.Context, evalCtx *eval.Context, args tree.Datums) (eval.ValueGenerator, error) {
-				isAdmin, err := evalCtx.SessionAccessor.HasAdminRole(ctx)
-				if err != nil {
+				if err := evalCtx.SessionAccessor.CheckPrivilege(
+					ctx, syntheticprivilege.GlobalPrivilegeObject, privilege.VIEWCLUSTERMETADATA,
+				); err != nil {
 					return nil, err
 				}
-				if !isAdmin {
```

Comment:
Does “view cluster metadata” really correspond to “read all data regardless of permissions”? I don’t have the written description of it in front of me but just by the name I assume it is eg SHOW RANGES or something not reading any kv?

Reply:
my thinking was that technically, you can see arbitrary KVs using SHOW RANGES, so this PR is not a privilege escalation

Reply:
We can leave out builtins like scan that can expose data from arbitrary key ranges. Barring that I would be okay with this.

Reply:
You can't see arbitrary KVs: you can see some keys if they were chosen as split keys, but you cannot see values. We already point out that indexed values end up in keys when we talk about domiciling or other data sensitivity questions, but _values_ are generally only visible to those granted access and admins.